metadata/contracts/DB3MetaStore.sol

```diff
@@ -1,15 +1,127 @@
+        registration.rollupNodeUrl = rollupNodeUrl;
+        registration.indexNodeUrls = indexNodeUrls;
+        registration.networkId = networkId;
+        registration.sender = msg.sender;
+        registration.rollupNodeAddress = rollupNodeAddress;
+
+        // Increment registered network counter
+        numNetworks++;
+    }
+
+    // Get registration info for a specific network ID
+    function getNetworkRegistration(uint64 networkId) public view returns (string memory rollupNodeUrl,address rollupNodeAddress, string[] memory indexNodeUrls, uint64 registrationNetworkId, address sender, bytes memory latestArweaveTx) {
+        // Get network registration struct and ensure it exists
+        NetworkRegistration storage registration = networkRegistrations[networkId];
+        require(bytes(registration.rollupNodeUrl).length > 0, "Network not registered");
+
+        // Return registration info
+        return (registration.rollupNodeUrl, registration.rollupNodeAddress,registration.indexNodeUrls, registration.networkId, registration.sender, registration.latestArweaveTx);
+    }
 
-import {IMetaStore} from "./interfaces/IMetaStore.sol";
+    // Get registration info for all networks (with pagination)
+    function getAllNetworkRegistrations(uint64 page, uint64 pageSize) public view returns (NetworkRegistration[] memory registrations) {
+        // Calculate number of registration infos to return
+        uint256 startIndex = (page - 1) * pageSize;
+        uint256 endIndex = startIndex + pageSize;
+        if (endIndex > numNetworks) {
+            endIndex = numNetworks;
+        }
+        uint256 numNetworksToReturn = endIndex - startIndex;
 
-contract DB3MetaStore is IMetaStore {
-    struct Network {
-        address admin;
-        address rollupNode;
-        address[] indexNodes;
-        bytes latestArTx;
+        // Create dynamic array to store registration infos
+        registrations = new NetworkRegistration[](numNetworksToReturn);
+
+        // Iterate over registered networks and add necessary registration infos to array
+        uint256 i = 0;
+        for (uint64 networkId = 1; networkId <= numNetworks; networkId++) {
+            if (bytes(networkRegistrations[networkId].rollupNodeUrl).length > 0) {
+                if ((i >= startIndex) && (i < endIndex)) {
+                    registrations[i - startIndex] = networkRegistrations[networkId];
+                }
+                i++;
+            }
+        }
+
+        // Return registration info array
+        return registrations;
+    }
+
+    // Register a new Rollup node for a specific network ID
+    function registerRollupNode(uint64 networkId, string memory rollupNodeUrl,address  rollupNodeAddress) public returns (bool success) {
+        // Check if network is registered
+        NetworkRegistration storage registration = networkRegistrations[networkId];
+        require(bytes(registration.rollupNodeUrl).length > 0, "Network not registered");
```

Comment:
1. only the network id and rollupNodeUrl are required
2. check the msg.sender == registration. rollupNodeAddress